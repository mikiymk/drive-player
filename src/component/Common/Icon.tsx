import React from "react";
import { css } from "@linaria/core";

const style = css`
  font-size: inherit;
  color: inherit;
`;

type Props = {
  icon: string;
  className?: string;
};

/** Google Material Icon */
const Icon: React.FC<Props> = ({ icon, className }) => {
  let classes = `material-icons-sharp ${style}`;
  if (className !== undefined) {
    classes = `material-icons-sharp ${style} ${className}`;
  }

  return <span className={classes}>{icon}</span>;
};

export default Icon;
