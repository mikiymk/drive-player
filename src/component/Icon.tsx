import React from "react";

const Icon: React.FC<{
  id: string;
}> = ({ id }) => <span className="material-icons-sharp">{id}</span>;

export default Icon;
