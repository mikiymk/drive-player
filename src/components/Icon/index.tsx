import { styleIcon } from "./style.css";

/* https://materialdesignicons.com/icon/play */
export const IconPlay = () => {
  return (
    <svg class={styleIcon} viewBox="0 0 24 24">
      <path d="M8,5.14V19.14L19,12.14L8,5.14Z" />
    </svg>
  );
};

/* https://materialdesignicons.com/icon/format-list-bulleted */
export const IconLibrary = () => {
  return (
    <svg class={styleIcon} viewBox="0 0 24 24">
      <path d="M7,5H21V7H7V5M7,13V11H21V13H7M4,4.5A1.5,1.5 0 0,1 5.5,6A1.5,1.5 0 0,1 4,7.5A1.5,1.5 0 0,1 2.5,6A1.5,1.5 0 0,1 4,4.5M4,10.5A1.5,1.5 0 0,1 5.5,12A1.5,1.5 0 0,1 4,13.5A1.5,1.5 0 0,1 2.5,12A1.5,1.5 0 0,1 4,10.5M7,19V17H21V19H7M4,16.5A1.5,1.5 0 0,1 5.5,18A1.5,1.5 0 0,1 4,19.5A1.5,1.5 0 0,1 2.5,18A1.5,1.5 0 0,1 4,16.5Z" />
    </svg>
  );
};

/* https://materialdesignicons.com/icon/playlist-music */
export const IconPlayList = () => {
  return (
    <svg class={styleIcon} viewBox="0 0 24 24">
      <path d="M15,6H3V8H15V6M15,10H3V12H15V10M3,16H11V14H3V16M17,6V14.18C16.69,14.07 16.35,14 16,14A3,3 0 0,0 13,17A3,3 0 0,0 16,20A3,3 0 0,0 19,17V8H22V6H17Z" />
    </svg>
  );
};

/* https://materialdesignicons.com/icon/google-drive */
export const IconDrive = () => {
  return (
    <svg class={styleIcon} viewBox="0 0 24 24">
      <path d="M7.71,3.5L1.15,15L4.58,21L11.13,9.5M9.73,15L6.3,21H19.42L22.85,15M22.28,14L15.42,2H8.58L8.57,2L15.43,14H22.28Z" />
    </svg>
  );
};

/* https://materialdesignicons.com/icon/google-drive */
export const IconSettings = () => {
  return (
    <svg class={styleIcon} viewBox="0 0 24 24">
      <path d="M12,15.5A3.5,3.5 0 0,1 8.5,12A3.5,3.5 0 0,1 12,8.5A3.5,3.5 0 0,1 15.5,12A3.5,3.5 0 0,1 12,15.5M19.43,12.97C19.47,12.65 19.5,12.33 19.5,12C19.5,11.67 19.47,11.34 19.43,11L21.54,9.37C21.73,9.22 21.78,8.95 21.66,8.73L19.66,5.27C19.54,5.05 19.27,4.96 19.05,5.05L16.56,6.05C16.04,5.66 15.5,5.32 14.87,5.07L14.5,2.42C14.46,2.18 14.25,2 14,2H10C9.75,2 9.54,2.18 9.5,2.42L9.13,5.07C8.5,5.32 7.96,5.66 7.44,6.05L4.95,5.05C4.73,4.96 4.46,5.05 4.34,5.27L2.34,8.73C2.21,8.95 2.27,9.22 2.46,9.37L4.57,11C4.53,11.34 4.5,11.67 4.5,12C4.5,12.33 4.53,12.65 4.57,12.97L2.46,14.63C2.27,14.78 2.21,15.05 2.34,15.27L4.34,18.73C4.46,18.95 4.73,19.03 4.95,18.95L7.44,17.94C7.96,18.34 8.5,18.68 9.13,18.93L9.5,21.58C9.54,21.82 9.75,22 10,22H14C14.25,22 14.46,21.82 14.5,21.58L14.87,18.93C15.5,18.67 16.04,18.34 16.56,17.94L19.05,18.95C19.27,19.03 19.54,18.95 19.66,18.73L21.66,15.27C21.78,15.05 21.73,14.78 21.54,14.63L19.43,12.97Z" />
    </svg>
  );
};

/* https://materialdesignicons.com/icon/log-in */
export const IconSignIn = () => {
  return (
    <svg class={styleIcon} viewBox="0 0 24 24">
      <path d="M10,17V14H3V10H10V7L15,12L10,17M10,2H19A2,2 0 0,1 21,4V20A2,2 0 0,1 19,22H10A2,2 0 0,1 8,20V18H10V20H19V4H10V6H8V4A2,2 0 0,1 10,2Z" />
    </svg>
  );
};

/* https://materialdesignicons.com/icon/log-out */
export const IconSignOut = () => {
  return (
    <svg class={styleIcon} viewBox="0 0 24 24">
      <path d="M16,17V14H9V10H16V7L21,12L16,17M14,2A2,2 0 0,1 16,4V6H14V4H5V20H14V18H16V20A2,2 0 0,1 14,22H5A2,2 0 0,1 3,20V4A2,2 0 0,1 5,2H14Z" />{" "}
    </svg>
  );
};

/* https://materialdesignicons.com/icon/file-music */
export const IconAudioFile = () => {
  return (
    <svg class={styleIcon} viewBox="0 0 24 24">
      <path d="M14,2H6A2,2 0 0,0 4,4V20A2,2 0 0,0 6,22H18A2,2 0 0,0 20,20V8L14,2M13,13H11V18A2,2 0 0,1 9,20A2,2 0 0,1 7,18A2,2 0 0,1 9,16C9.4,16 9.7,16.1 10,16.3V11H13V13M13,9V3.5L18.5,9H13Z" />
    </svg>
  );
};

/* https://materialdesignicons.com/icon/folder */
export const IconFolder = () => {
  return (
    <svg class={styleIcon} viewBox="0 0 24 24">
      <path d="M10,4H4C2.89,4 2,4.89 2,6V18A2,2 0 0,0 4,20H20A2,2 0 0,0 22,18V8C22,6.89 21.1,6 20,6H12L10,4Z" />
    </svg>
  );
};

/* https://materialdesignicons.com/icon/folder */
export const IconDotInfo = () => {
  return (
    <svg class={styleIcon} viewBox="0 0 24 24">
      <path d="M16,12A2,2 0 0,1 18,10A2,2 0 0,1 20,12A2,2 0 0,1 18,14A2,2 0 0,1 16,12M10,12A2,2 0 0,1 12,10A2,2 0 0,1 14,12A2,2 0 0,1 12,14A2,2 0 0,1 10,12M4,12A2,2 0 0,1 6,10A2,2 0 0,1 8,12A2,2 0 0,1 6,14A2,2 0 0,1 4,12Z" />
    </svg>
  );
};

/* https://materialdesignicons.com/icon/download */
export const IconDownload = () => {
  return (
    <svg class={styleIcon} viewBox="0 0 24 24">
      <path d="M5,20H19V18H5M19,9H15V3H9V9H5L12,16L19,9Z" />
    </svg>
  );
};

/* https://materialdesignicons.com/icon/upload */
export const IconUpload = () => {
  return (
    <svg class={styleIcon} viewBox="0 0 24 24">
      <path d="M9,16V10H5L12,3L19,10H15V16H9M5,20V18H19V20H5Z" />
    </svg>
  );
};

/* https://materialdesignicons.com/icon/dots-horizontal-circle-outline */
export const IconLoading = () => {
  return (
    <svg class={styleIcon} viewBox="0 0 24 24">
      <path d="M12,2A10,10 0 0,1 22,12A10,10 0 0,1 12,22A10,10 0 0,1 2,12A10,10 0 0,1 12,2M12,4A8,8 0 0,0 4,12A8,8 0 0,0 12,20A8,8 0 0,0 20,12A8,8 0 0,0 12,4M12,10.5A1.5,1.5 0 0,1 13.5,12A1.5,1.5 0 0,1 12,13.5A1.5,1.5 0 0,1 10.5,12A1.5,1.5 0 0,1 12,10.5M7.5,10.5A1.5,1.5 0 0,1 9,12A1.5,1.5 0 0,1 7.5,13.5A1.5,1.5 0 0,1 6,12A1.5,1.5 0 0,1 7.5,10.5M16.5,10.5A1.5,1.5 0 0,1 18,12A1.5,1.5 0 0,1 16.5,13.5A1.5,1.5 0 0,1 15,12A1.5,1.5 0 0,1 16.5,10.5Z" />
    </svg>
  );
};

/* https://materialdesignicons.com/icon/check-circle-outline */
export const IconDone = () => {
  return (
    <svg class={styleIcon} viewBox="0 0 24 24">
      <path d="M12 2C6.5 2 2 6.5 2 12S6.5 22 12 22 22 17.5 22 12 17.5 2 12 2M12 20C7.59 20 4 16.41 4 12S7.59 4 12 4 20 7.59 20 12 16.41 20 12 20M16.59 7.58L10 14.17L7.41 11.59L6 13L10 17L18 9L16.59 7.58Z" />
    </svg>
  );
};

/* https://materialdesignicons.com/icon/alert-circle-outline */
export const IconError = () => {
  return (
    <svg class={styleIcon} viewBox="0 0 24 24">
      <path d="M11,15H13V17H11V15M11,7H13V13H11V7M12,2C6.47,2 2,6.5 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20Z" />
    </svg>
  );
};

/* https://materialdesignicons.com/icon/skip-previous */
export const IconSkipPrev = () => {
  return (
    <svg class={styleIcon} viewBox="0 0 24 24">
      <path d="M6,18V6H8V18H6M9.5,12L18,6V18L9.5,12Z" />
    </svg>
  );
};

/* https://materialdesignicons.com/icon/skip-next */
export const IconSkipNext = () => {
  return (
    <svg class={styleIcon} viewBox="0 0 24 24">
      <path d="M16,18H18V6H16M6,18L14.5,12L6,6V18Z" />
    </svg>
  );
};

/* https://materialdesignicons.com/icon/pause */
export const IconPause = () => {
  return (
    <svg class={styleIcon} viewBox="0 0 24 24">
      <path d="M14,19H18V5H14M6,19H10V5H6V19Z" />
    </svg>
  );
};

/* https://materialdesignicons.com/icon/repeat */
export const IconRepeatOn = () => {
  return (
    <svg class={styleIcon} viewBox="0 0 24 24">
      <path d="M17,17H7V14L3,18L7,22V19H19V13H17M7,7H17V10L21,6L17,2V5H5V11H7V7Z" />
    </svg>
  );
};

/* https://materialdesignicons.com/icon/repeat-off */
export const IconRepeatOff = () => {
  return (
    <svg class={styleIcon} viewBox="0 0 24 24">
      <path d="M2,5.27L3.28,4L20,20.72L18.73,22L15.73,19H7V22L3,18L7,14V17H13.73L7,10.27V11H5V8.27L2,5.27M17,13H19V17.18L17,15.18V13M17,5V2L21,6L17,10V7H8.82L6.82,5H17Z" />
    </svg>
  );
};

/* https://materialdesignicons.com/icon/repeat-once */
export const IconRepeatOnce = () => {
  return (
    <svg class={styleIcon} viewBox="0 0 24 24">
      <path d="M13,15V9H12L10,10V11H11.5V15M17,17H7V14L3,18L7,22V19H19V13H17M7,7H17V10L21,6L17,2V5H5V11H7V7Z" />
    </svg>
  );
};

/* https://materialdesignicons.com/icon/shuffle */
export const IconShuffleOn = () => {
  return (
    <svg class={styleIcon} viewBox="0 0 24 24">
      <path d="M14.83,13.41L13.42,14.82L16.55,17.95L14.5,20H20V14.5L17.96,16.54L14.83,13.41M14.5,4L16.54,6.04L4,18.59L5.41,20L17.96,7.46L20,9.5V4M10.59,9.17L5.41,4L4,5.41L9.17,10.58L10.59,9.17Z" />
    </svg>
  );
};

/* https://materialdesignicons.com/icon/shuffle-disabled */
export const IconShuffleOff = () => {
  return (
    <svg class={styleIcon} viewBox="0 0 24 24">
      <path d="M16,4.5V7H5V9H16V11.5L19.5,8M16,12.5V15H5V17H16V19.5L19.5,16" />
    </svg>
  );
};

/* https://materialdesignicons.com/icon/close */
export const IconClose = () => {
  return (
    <svg class={styleIcon} viewBox="0 0 24 24">
      <path d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z" />
    </svg>
  );
};
