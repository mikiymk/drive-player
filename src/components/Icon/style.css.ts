import { style } from "@vanilla-extract/css";

export const styleIcon = style({
  fill: "currentcolor",
  height: "1em",
  width: "1em",
  verticalAlign: "sub",
});
