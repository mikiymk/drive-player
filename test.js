(() => {
  const performance = require("perf_hooks").performance;
  let marr = length => Array.from({ length }).map((_, i) => ["s_" + i, i * i]);
  let mmap = l => new Map(marr(l));
  let len = 1000;
  let cnt = 10000;

  let test = (label, fn) => {
    let times = [];
    for (let i = 0; i < cnt; i++) {
      let t0 = performance.now();
      fn(mmap(len), marr(len));
      let t1 = performance.now();
      times.push(t1 - t0);
    }
    let ave = times.reduce((p, c) => p + c) / cnt;
    let vari = times.reduce((p, c) => p + (c - ave) ** 2) / cnt;
    let err = Math.sqrt(vari / (cnt - 1)) * 100;

    return `${label} ${ave.toFixed(4)}Â±${err.toFixed(2)}%`;
  };

  let a;
  a = test("foreach", (m, a) => {
    var n = new Map(m);
    a.forEach(aa => n.set(aa[0] + aa[0], aa[1] + 1));
    return n;
  });
  console.log(a);

  a = test("map    ", (m, a) => {
    var n = new Map([...m, ...a.map(aa => [aa[0] + aa[0], aa[1] + 1])]);
    return n;
  });
  console.log(a);

  a = test("for in ", (m, a) => {
    var n = new Map(m);
    for (let i in a) {
      let aa = a[i];
      n.set(aa[0] + aa[0], aa[1] + 1);
    }
    return n;
  });
  console.log(a);

  a = test("for of ", (m, a) => {
    var n = new Map(m);
    for (let aa of a) {
      n.set(aa[0] + aa[0], aa[1] + 1);
    }
    return n;
  });
  console.log(a);

  a = test("for i  ", (m, a) => {
    var n = new Map(m);
    for (let i = 0; i < a.length; i++) {
      let aa = a[i];
      n.set(aa[0] + aa[0], aa[1] + 1);
    }
    return n;
  });
  console.log(a);
})();
