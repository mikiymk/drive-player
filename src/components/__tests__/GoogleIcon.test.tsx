import { test, expect, describe } from "vitest";
import { mockCSS, render } from "~/test/helper";

import Icon from "../GoogleIcon";

describe("<Icon />", () => {
  mockCSS();

  test("icon component test", () => {
    const { container, unmount } = render(() => <Icon icon="icon" />);

    expect(container.innerHTML).toMatchSnapshot();

    unmount();
  });
});
