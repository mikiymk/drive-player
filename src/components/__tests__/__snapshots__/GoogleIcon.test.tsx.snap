// Vitest Snapshot v1

exports[`<Icon /> > icon component test 1`] = `"<span class=\\"material-icons-sharp style__t4deip0\\">icon</span>"`;
